metricbeat/module/mysql/query/query.go

```diff
@@ -41,18 +40,19 @@ func init() {
 }
 
 type query struct {
-	QueryNamespace string `config:"query_namespace"`
-	Query          string `config:"query"`
+	Namespace      string `config:"query_namespace" validate:"nonzero,required"`
```

Comment:
The problem of forcing the use of a custom namespace in a generic module is that we cannot provide default mappings as we do in the sql module with the `sql.metrics...` fields.
We should probably use some namespace by default, and provide a mapping for it.

Reply:
I'll open an issue to add default mappings to `query` metricset because they can use `sql` module as a workaround. This way I don't make this PR even bigger.